spec/test_app/bin/setup

```
@@ -1,29 +1,29 @@
 #!/usr/bin/env ruby
-require 'pathname'
+require "fileutils"
 
-# path to your application root.
-APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
+APP_ROOT = File.expand_path("..", __dir__)
+APP_NAME = "test-app"
 
-Dir.chdir APP_ROOT do
-  # This script is a starting point to setup your application.
-  # Add necessary setup steps to this file:
-
-  puts "== Installing dependencies =="
-  system "gem install bundler --conservative"
-  system "bundle check || bundle install"
+def system!(*args)
+  system(*args, exception: true)
+end
 
-  # puts "\n== Copying sample files =="
-  # unless File.exist?("config/database.yml")
-  #   system "cp config/database.yml.sample config/database.yml"
-  # end
+FileUtils.chdir APP_ROOT do
+  # This script is a way to set up or update your development environment automatically.
+  # This script is idempotent, so that you can run it at any time and get an expectable outcome.
+  # Add necessary setup steps to this file.
 
-  puts "\n== Preparing database =="
-  system "bin/rake db:setup"
+  puts "== Installing dependencies =="
+  system! "gem install bundler --conservative"
+  system("bundle check") || system!("bundle install")
```

Is there any reason to avoid `system!` for the bundle check?

Appreciate the decision to make this fail fast, seems like a good call, just also seems like a win to fully invest in that good choice unless I'm missing something.

---

Since this is just a dummy app for running specs, I stuck with what rails app:update generated — this wasn’t something I intentionally added; it's part of the default setup file in Rails 7.2.